client/core/core.go

```diff
@@ -67,6 +67,7 @@ type dexConnection struct {
 	marketMtx sync.RWMutex
 	marketMap map[string]*Market
 
+	submittingTrades sync.WaitGroup
```

Comment:
This is a workaround to the unfortunate possibility that the server will send a preimage request before the client receives, validates, and stores the order response as a trackedTrade in dc.trades.  We dealt with this previously by holding `dc.tradeMtx` locked for the duration of the request through storing it in the map, but that is not feasible anymore with a potentially long new order ('limit' or 'market') request.

The `Add`/`Done` occurs [here](https://github.com/decred/dcrdex/pull/688/files#diff-5c4d3ee4527d6c50d1379b8edb5ef11eR2235-R2236).
The possible `Wait` occurs [here](https://github.com/decred/dcrdex/pull/688/files#diff-5c4d3ee4527d6c50d1379b8edb5ef11eR3464-R3473), but only if the preimage request handler fails to find the order id.